lib/types/inet-address.js

```diff
@@ -10,238 +11,243 @@ const utils = require("../utils");
+            return new InetAddress(utils.allocBufferFromArray(parts));
+        }
+        if (!ipv6Pattern.test(value)) {
+            throw new TypeError(
+                "Value could not be parsed as InetAddress: " + value,
+            );
         }
-        //its an empty string
-        if (applied) {
-            //there could be 2 occurrences of empty string
-            filling = 1;
+        parts = value.split(":");
+        if (parts.length < 3) {
+            throw new TypeError(
+                "Value could not be parsed as InetAddress: " + value,
+            );
         }
-        applied = true;
-        for (let j = 0; j < filling; j++) {
-            buffer[offset++] = 0;
-            buffer[offset++] = 0;
+        const buffer = utils.allocBufferUnsafe(16);
+        let filling = 8 - parts.length + 1;
+        let applied = false;
+        let offset = 0;
+        const embeddedIp4 = ipv4Pattern.test(parts[parts.length - 1]);
+        if (embeddedIp4) {
+            // Its IPv6 address with an embedded IPv4 address:
+            // subtract 1 from the potential empty filling as ip4 contains 4 bytes instead of 2 of a ipv6 section
+            filling -= 1;
+        }
+        function writeItem(uIntValue) {
+            buffer.writeUInt8(+uIntValue, offset++);
+        }
+        for (let i = 0; i < parts.length; i++) {
+            const item = parts[i];
+            if (item) {
+                if (embeddedIp4 && i === parts.length - 1) {
+                    item.split(".").forEach(writeItem);
+                    break;
+                }
+                buffer.writeUInt16BE(parseInt(item, 16), offset);
+                offset = offset + 2;
+                continue;
+            }
+            //its an empty string
+            if (applied) {
+                //there could be 2 occurrences of empty string
+                filling = 1;
+            }
+            applied = true;
+            for (let j = 0; j < filling; j++) {
+                buffer[offset++] = 0;
+                buffer[offset++] = 0;
+            }
         }
+        if (embeddedIp4 && !isValidIPv4Mapped(buffer)) {
+            throw new TypeError(
+                "Only IPv4-Mapped IPv6 addresses are allowed as IPv6 address with embedded IPv4 address",
+            );
+        }
+        return new InetAddress(buffer);
     }
-    if (embeddedIp4 && !isValidIPv4Mapped(buffer)) {
-        throw new TypeError(
-            "Only IPv4-Mapped IPv6 addresses are allowed as IPv6 address with embedded IPv4 address",
+
+    /**
+     * Compares 2 addresses and returns true if the underlying bytes are the same
+     * @param {InetAddress} other
+     * @returns {Boolean}
+     */
+    equals(other) {
+        if (!(other instanceof InetAddress)) {
+            return false;
+        }
+        return (
+            this.buffer.length === other.buffer.length &&
+            this.buffer.toString("hex") === other.buffer.toString("hex")
```

Comment:
You are comparing attributes that do not exist anymore.

Reply:
IIUC, this is exactly what we want! We emulate reading attributes by use of `get`ters. Thanks to that, the API is unchanged wrt attribute reads; only writes are no longer valid. BTW, what happens if an emulated attribute (i.e., a name with a `get`ter but without a `set`ter) is written to?

Reply:
Nothing happens, no error thrown, no value changed